pkg/serviceprovider/github/githubclientbuilder.go

```diff
@@ -33,6 +35,14 @@ type githubClientBuilder struct {
 	tokenStorage tokenstorage.TokenStorage
 }
 
+func (g *githubClientBuilder) CreateAuthenticatedClient(ctx context.Context, credentials serviceprovider.Credentials) (*github.Client, error) {
```

Comment:
What's the idea of using pointer type recievers there?

Reply:
yes, good point. I will refactor it for the main types like `Github` and `Gitlab`, but not for structs like `downloadFileCapability` as that calls a non-interfaced function (`createGitlabAuthClient`) and I do not want to refactor it in this PR.